src/Ldap.php

```diff
@@ -923,17 +923,16 @@ public function search($filter, $basedn = null, $scope = self::SEARCH_SCOPE_SUB,
         if ($search === false) {
             throw new Exception\LdapException($this, 'searching: ' . $filter);
         }
+
+        $iterator = new Collection\SortableIterator($this, $search);
+
         if ($sort !== null && is_string($sort)) {
-            ErrorHandler::start(E_WARNING);
-            $isSorted = ldap_sort($resource, $search, $sort);
-            ErrorHandler::stop();
+            $isSorted = $iterator->sort($sort);
             if ($isSorted === false) {
                 throw new Exception\LdapException($this, 'sorting: ' . $sort);
```

Comment:
Why is this exception required?

If required I think can be placed in sort() method